### PR TITLE
Scuttlego in Planetary

### PR DESCRIPTION
**Pull request by boreq**
This is a build of Planetary which uses [scuttlego](https://github.com/planetary-social/scuttlego) under the hood. Initially when scuttlego didn't have a name we jokingly referred to it as "Project Raptor" so that it sounds cool and mysterious, hence the name of this branch. I am opening this pull request for visibility so that everyone can see that this branch exists.

Relevant:
- [Main planetary project board](https://github.com/orgs/planetary-social/projects/5/views/4)
- [Milestone in planetary-social/planetary-ios](https://github.com/planetary-social/planetary-ios/milestone/7)
- [Milestone in planetary-social/scuttlego](https://github.com/planetary-social/scuttlego/milestone/1)

Bindings:
- [x] ssbBotStop
- [x] ssbBotIsRunning
- [x] ssbBotInit
- [x] ssbBlobsWant 
- [x] ssbBlobsAdd *note: needs to push blobs?*
- [x] ssbConnectPeer
- [x] ssbDisconnectAllPeers
- [x] ssbFeedReplicate *note: used only to always mark own identity + service pubs as feeds to replicate, changed to no longer accept the second parameter (`yes`)*
- [ ] ssbFeedBlock *note: unclear, needs feedback on slack* https://github.com/planetary-social/planetary-ios/issues/1008
- [x] ssbNullContent *note: removed, should be no longer needed - banning/unbanning/blocking/unblocking should remove the feed automatically*
- [x] ssbInviteAccept (bug? pubs say "invite.use" is not in the list of allowed methods)
- [x] ssbRoomsAliasRegister
- [x] ssbRoomsAliasRevoke
- [x] ssbRoomsListAliases
- [x] ssbBotStatus *note: returns fake pid, blobs, root, indices* 
- [x] ssbOpenConnections 
- [x] ssbRepoStats *note: returns fake last hash*
- [x] ssbPublish
- [x] ssbPublishPrivate *note: noop*
- [x] ssbGetRawMessage
- [x] ssbStreamRootLog
- [x] ssbStreamPrivateLog *note: noop*
- [x] ssbStreamPublishedLog 
- [x] ssbTestingMakeNamedKey 
- [x] ssbTestingAllNamedKeypairs 
- [x] ssbTestingPublishAs 
- [x] ssbTestingPublishPrivateAs *note: private messages are not supported for now*
- [x] ssbGenKey *note: still uses go-ssb directly for no paritcular reason?*
- [x] ssbOffsetFSCK *todo: remove?* https://github.com/planetary-social/planetary-ios/issues/969
- [x] ssbHealRepo *todo: remove?* https://github.com/planetary-social/planetary-ios/issues/969
- [x] ssbConnectPeers *removed as this function was unused*
- [x] planetaryBearerToken *removed as this function was unused*
- [x] ssbReplicateUpTo *removed as this function was unused*
- [x] ssbDropIndexData *removed as this function no longer serves any purpose*

### CODE REVIEW

**Comment by boreq** [comment]
Most tests are passing now except for a couple of published log tests.

**Comment by boreq** [comment]
Relevant issues:
- https://github.com/planetary-social/planetary-ios/issues/1007
- https://github.com/planetary-social/planetary-ios/issues/1006

**Comment by martindsq** [comment]
@boreq you can now merge/rebase main to see if unit tests pass in this branch

**Comment by boreq** [comment]
I added Github actions for Go bindings here which I think will be useful. For now we don't have a lot of tests but I want to add them and make sure that a passing pipeline is required to merge changes to `main`.

**Comment by boreq** [comment]
Let's go over all the changes again and clean up before the merge now.

File: UnitTests/GoBotOrderedTests.swift [hunk 1]
```diff
@@ -214,7 +214,7 @@ class GoBotOrderedTests: XCTestCase {
         XCTAssertEqual(publishedMessages.last?.contentType, .about)
     }
     
-    /// Tests that the getPublishedLog function returns nil when we pass an out-of-bounds index.
+    /// Tests that the getPublishedLog function throws an error when we pass an out-of-bounds index.
```
Comment (boreq) [inline UnitTests/GoBotOrderedTests.swift:217]:
```suggestion
    /// Tests that the getPublishedLog returns an empty list when we pass an out-of-bounds index.
```

File: Support/.swiftpm/xcode/xcshareddata/xcschemes/Support.xcscheme [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1410"
+   LastUpgradeVersion = "1400"
```
Comment (boreq) [inline Support/.swiftpm/xcode/xcshareddata/xcschemes/Support.xcscheme:3]: What are all those changes? Is this something we should commit?

File: Source/GoBot/GoBotInternal.swift [hunk 3]
```diff
@@ -411,34 +448,22 @@ class GoBotInternal {
     
     /// Instructs the bot to stop replicating the current feed without publishing a message on the user's log.
     func ban(feed: FeedIdentifier) {
-        feed.withGoString {
-            ssbFeedBlock($0, true)
-        }
+//        feed.withGoString {
```
Comment (boreq) [inline Source/GoBot/GoBotInternal.swift:451]: There is a ticket for this.

File: Planetary.xcodeproj/xcshareddata/xcschemes/Performance Tests.xcscheme [hunk 4]
```diff
@@ -5,31 +5,39 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <BuildActionEntries>
```
Comment (boreq) [inline Planetary.xcodeproj/xcshareddata/xcschemes/Performance Tests.xcscheme:8]: ???

File: UnitTests/Performance Tests.xctestplan [hunk 5]
```diff
@@ -14,6 +14,7 @@
   "testTargets" : [
     {
       "selectedTests" : [
+        "GoBotIntegrationTests\/testPublishPerformance()",
```
Comment (boreq) [inline UnitTests/Performance Tests.xctestplan:17]: Is this correct?

**Comment by boreq** [comment]
Created:
- https://github.com/planetary-social/planetary-ios/issues/1157
- https://github.com/planetary-social/planetary-ios/issues/1156
- https://github.com/planetary-social/planetary-ios/issues/1152
- https://github.com/planetary-social/planetary-ios/issues/1155
- https://github.com/planetary-social/planetary-ios/issues/1153
- https://github.com/planetary-social/planetary-ios/issues/1154
- https://github.com/planetary-social/planetary-ios/issues/1158

**Comment by boreq** [comment]
I can't sign off on my own pull request but the Go changes have all been reviewed and seem reasonable. I also read the Swift code and the project files and while I don't entirely understand those they seem ok as well.

**Review by martindsq — APPROVED** [review]
Most of the changes from the swift side were already reviewed in the migration UI PR, I checked the remaining ones and everything is in place

**Comment by boreq** [comment]
Can we fix Codacy to be 100% green?

**Pull request merged**